src/io/flutter/logging/FlutterConsoleLogManager.java

```diff
@@ -144,4 +172,54 @@ private void printStackTraceToConsole(
   private String stringValueFromStringRef(InstanceRef ref) {
     return ref.getValueAsStringIsTruncated() ? ref.getValueAsString() + "..." : ref.getValueAsString();
   }
+
+  private String stringValueFromStringRef(Instance instance) {
+    return instance.getValueAsStringIsTruncated() ? instance.getValueAsString() + "..." : instance.getValueAsString();
+  }
+
+  private String getFullStringValue(@NotNull VmService service, String isolateId, @Nullable InstanceRef ref) {
+    if (ref == null) return null;
+
+    if (!ref.getValueAsStringIsTruncated()) {
+      return ref.getValueAsString();
+    }
+
+    final CountDownLatch latch = new CountDownLatch(1);
+    final String[] result = new String[1];
+
+    service.getObject(isolateId, ref.getId(), 0, ref.getLength(), new GetObjectConsumer() {
+      @Override
+      public void onError(RPCError error) {
+        result[0] = ref.getValueAsString() + "...";
+        latch.countDown();
+      }
+
+      @Override
+      public void received(Obj response) {
+        if (response instanceof Instance && ((Instance)response).getKind() == InstanceKind.String) {
```

Comment:
Does it need to be cast to an `Instance` if it passes the `instanceof` check?

Reply:
It does, through I could also assign it to a new local variable.